marketplace/src/app/model/Product.ts

```diff
@@ -0,0 +1,17 @@
+export class Product {
+  title: string
+  price: number
+  description: string
+  vendorId: number
+  attributes: Array<Pair>
+
+  constructor() {
+    this.attributes = new Array<Pair>()
+  }
+
+}
+
+export class Pair {
```

Comment:
Move the pairs to a separate file and give them a more specific name (AttributePair or AttributeValue maybe?)